### ISSUE TITLE
Books on writing CLI/TUI apps

### ISSUE THREAD

**Issue by wmealing** [agarrharr/awesome-cli-apps#541]
I don't know if there is any, but a section on designing usable TUI apps would be nice to have.  I don't know of any and my search on the topic had lead me here.

If you know of any, this might be a good place to put links to them.

**Comment by jneidel** [comment]
I'm gonna be writing my bachelors thesis on something along those lines soon. I could add a section with my sources and research to the list afterwards :smile:

**Comment by wmealing** [comment]
Any chance that was finished.   I am still keen on reading up on the topic?

**Comment by Da-Buche** [comment]
It's not exactly the answer you look for.

But if you know Python and want to make neat TUI apps, this might be a very good place to start:
https://textual.textualize.io/
It's open-source, here's the associated repo : https://github.com/textualize/textual/

You can try the demo using pip :
```sh
python -m pip install textual
python -m textual
```

Edit: There is also a dedicated book called 'Creating TUI Applications with Textual and Python'

**Comment by wmealing** [comment]
Not quite ! but thank you anyway.  I didnt know about the book, thats a great start.

**Comment by jneidel** [comment]
> Any chance that was finished. I am still keen on reading up on the topic?

@wmealing 
Yes, was done 2 years ago. You can find it [here](https://github.com/jneidel/ba). It is in german, but the methods I used I listed in english as well. You can also load it into chatGPT and ask questions about it in english (or for a translation, etc.)
